modules/principals/src/main/frontend/src/Userboard/Groups/addusertogroupdialogue.jsx

```diff
@@ -0,0 +1,181 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+*/
+
+import React from "react";
+
+import { withTheme } from "@material-ui/core/styles";
+
+import { Button, Dialog, DialogTitle, DialogActions, DialogContent, Grid, Table, TableBody, TableHead, TableRow, TableCell } from "@material-ui/core";
+
+import MaterialTable from 'material-table';
+
+const GROUP_URL="/system/userManager/group/";
+
+class AddUserToGroupDialogue extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            allUserNames: [],
+            freeUserNames: [],
+            selectedUsers: []
+        }
+    }
+
+    addName(name) {
+        return { name }
+    }
+
+    handleLoadUsers() {
+        fetch("/system/userManager/user.1.json",
+            {
+                method: 'GET',
+                headers: {
+                    'Authorization': 'Basic ' + btoa('admin:admin')
+                }
+            })
+            .then((response) => {
+                return response.json();
+            })
+            .then((data) => {
+                var names = [];
+                for (var username in data) {
+                    names = Object.keys(data).filter((username) => !username.startsWith("sling-")).map((username) => this.addName(username));
+                }
```

Comment:
The for loop surrounding line 56 looks like a leftover, remove.

```suggestion
                var names = Object.keys(data).filter((username) => !username.startsWith("sling-")).map((username) => this.addName(username));
```